File: backend/src/main/java/com/whiteblue/backend/domain/board/entity/Board.java
package com.whiteblue.backend.domain.board.entity;

import com.whiteblue.backend.domain.user.entity.User;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
@Entity
public class Board {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotBlank
    @Size(max = 30)
    private String title;

    @NotBlank
    private String content;

    @CreatedDate
    private LocalDateTime createdDate;

    @LastModifiedDate
    private LocalDateTime modifiedDate;

    @NotNull
    @ManyToOne(cascade = CascadeType.PERSIST, fetch = FetchType.LAZY)
    private User writer;

    @Builder
    public Board(Integer id, String title, String content, User writer) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.writer = writer;
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/board/Board.kt
package com.whiteblue.backend.domain.board

import com.whiteblue.backend.domain.user.User
import com.whiteblue.backend.util.BaseTimeEntity
import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank

@Table(name = "board")
@Entity
class Board(
    @field:NotBlank
    var title: String,

    @field:NotBlank
    var content: String,

    @ManyToOne(cascade = [CascadeType.PERSIST], fetch = FetchType.LAZY)
    @JoinColumn(name = "writer_id")
    val writer: User
) : BaseTimeEntity() {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    val id: Long? = null
}
